src/utils.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ *
+ *
+ * @param name
+ */
+export async function validateScheduledSurfaceGuid(
+  name: string
+): Promise<void> {
+  const allowlist = ['POCKET_HITS_EN_US', 'POCKET_HITS_DE_DE'];
+
+  if (allowlist.includes(name)) {
+    return;
+  } else {
+    throw new Error('Not a valid Scheduled Surface');
+  }
+}
+
+/**
+ *
+ * @param date
+ */
+export async function validateDate(date: string): Promise<void> {
+  const regEx = /^\d{4}-\d{2}-\d{2}$/;
+
+  if (date.match(regEx) === null) {
+    throw new Error(
+      'Not a valid date. Please provide a date in YYYY-MM-DD format.'
+    );
+  }
+
+  return;
```

Comment:
do these need to be `async`?

Reply:
also what do we think about doing the `Sentry` call directly in these functions? would save us the `try` blocks in the other file...

Reply:
@jpetto, I'd love some guidance on when we should call Sentry and what we should log there. When talking this through with @bassrock last week we realised most errors here should actually be exposed for Braze to know the endpoint is not returning data so that it doesn't send out emails with no Pocket Hits in them...

Reply:
Also, the utils definitely don't need to be async, not sure what came over me to make them that way. Removed - they're super simple now. Do you think we need to make sure the date provided is a valid date (i.e. not `0000-00-00` or `1111-33-55`)? Or just checking for the date in the right format is enough?

Reply:
agree with daniel above. there are probably a couple cases where we definitely _do_ want to pass through the error to braze, specifically malformed input data (scheduled surface GUID and date), and no results. any other error should be masked and return a generic 500 internal server error.

i would say _all_ errors should be sent to Sentry prior to being masked.

we could potentially define to new error classes here that inherit from `Error` - something like `UserInputError` for malformed params, and `NotFoundError` for missing data. in the error handler, if the error is either one of those, we pass it through to braze. otherwise, we mask it to a generic 500.

also not sure on how to do this an "Express" way - but you're probably halfway there with the `use` statement in this PR 😄

Reply:
for express apps, we can use the http-errors https://www.npmjs.com/package/http-errors as middleware. we can set the message and error type at controller/service layer and the middleware would format and propogate them to the client

Reply:
I have added comments in the code and a link to Braze docs to explain this, thank you Sri!